src/qibo/optimizers.py

```diff
@@ -56,8 +56,16 @@ def newtonian(loss, initial_parameters, method='Powell', options=None):
         options (dict): Dictionary with options accepted by
             ``scipy.optimize.minimize``.
     """
-    from scipy.optimize import minimize
-    m = minimize(loss, initial_parameters, method=method, options=options)
+    if method == 'parallel_lbfgsb':
+        from qibo.config import raise_error, get_backend, THREADS
+        if get_backend() == 'custom' and THREADS.get("intra") != 1:
+            raise_error(RuntimeError, "Parallel L-BFGS-B with custom operator requires to"
+                                    " set the QIBO_NUM_THREADS=1 environment variable.")
+        from optimparallel import minimize_parallel
```

Comment:
I would add an additional check
```Python
if "GPU" in get_device():
    raise_error(RuntimeError, "Parallel L-BFGS-B cannot be used with GPU.")
```
The reason is that when I first tested this on my machine, I accidentally forgot the GPU enabled and noticed that the code just gets stuck and does nothing in this case. In addition to this, if the script is killed with Ctrl+C then the GPU memory is not emptied (same problem we observed on DGX) unless the machine is rebooted (which can be annoying and may even not be possible for some servers).